app/src/ui/app.tsx

```diff
@@ -1177,6 +1200,11 @@ export class App extends React.Component<IAppProps, IAppState> {
     this.props.dispatcher.clearError(error)
   }
 
+  private onConfirmDiscardChangesChanged = (value: boolean) => {
+    this.setState({ askForConfirmationOnDiscardChanges: value })
```

Comment:
This shouldn't be necessary. The app store will emit an update and the app will grab the updated state.

Reply:
So no state update?! 😢

Reply:
Right, I don't think we should ever need to update `App`s state manually.